plugins/module_utils/mso.py

```diff
@@ -853,6 +856,13 @@ def lookup_roles(self, roles, ignore_not_found_error=False):
             ids.append(dict(roleId=r.get("id"), accessType=access_type))
         return ids
 
+    def lookup_site_type(self, s):
+        """Get site type(AWS, AZURE or physical)"""
+        platform = s.get("platform")
+        if platform == "cloud":
+            self.platform = s.get("cloudProviders")[0]
+        self.platform = platform
```

Comment:
Should we use self.platform for this? Since the platform is used to determine the connection plugin that should be used.